mito-ai/src/components/AttachFileButton.tsx

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) Saga Inc.
+ * Distributed under the terms of the GNU Affero General Public License v3.0 License.
+ */
+
+import React, { useRef, useState } from 'react';
+import IconButton from './IconButton';
+import PaperClipIcon from '../icons/PaperClipIcon';
+import { requestAPI } from '../restAPI/utils';
+
+interface AttachFileButtonProps {
+    onFileUploaded: (fileName: string) => void;
+}
+
+const AttachFileButton: React.FC<AttachFileButtonProps> = ({ onFileUploaded }) => {
+    const fileInputRef = useRef<HTMLInputElement>(null);
+    const [isUploading, setIsUploading] = useState(false);
+
+    // Constants for file handling
+    const MAX_FILE_SIZE = 100 * 1024 * 1024; // 100MB in bytes
+    const CHUNK_SIZE = 50 * 1024 * 1024; // 50MB chunks
```

Comment:
A few small things here: 

0. I think Jupyter uses much smaller max_file_size and chunk sizes. If I were to upload a file that is 25MB I think I would get the popup that says "This is a large file are you sure you want to upload". Want to look at the Jupyter code and just use the same cutoffs + chink sizes since they are presumably very well tested.
1. It looks like this isn't a max file size it's just the cutoff for if we treat it as chunks or not. Max file size made me think we reject files larger than this. Can we just update the name?
2. Can we just move these contents to the top of the file outside of the component.